Files/ViewModels/SettingsViewModel.cs

```diff
@@ -607,6 +607,15 @@ public bool UseFileListCache
             set => Set(value);
         }
 
+        /// <summary>
+        /// Gets or sets a value indicating the limit of parallel preemptive cache loading limit.
+        /// </summary>
+        public int PreemptiveCacheParallelLimit
+        {
+            get => Get(5);
```

Comment:
Enabled by default is ok @yaichenbaum?

Reply:
That's fine, after testing it out, I think we should lower it a bit since it's a little heavy on system resources.

Reply:
I think it might be good to consider different values for different drive type

Reply:
> I think it might be good to consider different values for different drive type

That might be a good way to do it, I would also consider changing this option into an on/off switch instead of letting the user choose the actual value.

Reply:
Do you have some tested way to check for the drive type in UWP?

Reply:
@jakoss Not that I know of, perhaps we should stick with a lower number for now and if we find a way to detect the drive type, we can adjust it later on.

Reply:
Ok, i lowered this to 2 (i think it's better to target hard drives for now)

Reply:
I adjusted the settings (new on/off switch). I think PR is ready for review and merge

Reply:
Should we remove the setting to adjust the limit and just have the option to turn it on or off?

Reply:
I think it's fine the way it is now. On/off switch is independent of limit setting. Limit selector will be enabled only if caching and preemptive caching are on. Users with fast ssd drives can bump this up. Minimum is 1